### PR TITLE
feat: Support RFC9457 response formats

### PR DESCRIPTION
**Pull request by EdgyEdgemond**
`generate_handler(legacy=True)` to continue generating old style responses.

Refs: #15

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,27 +6,40 @@
 ![tests](https://github.com/EdgyEdgemond/web-error/workflows/tests/badge.svg)
 [![codecov](https://codecov.io/gh/EdgyEdgemond/web-error/branch/master/graph/badge.svg)](https://codecov.io/gh/EdgyEdgemond/web-error)
 
-`web_error` is a set of exceptions and handlers for use in web apis to support easy error management and responses
-
-Each exception easily marshals to JSON for use in api errors. Handlers for different web frameworks are provided.
+`web_error` is a set of exceptions and handlers for use in starlette/fastapi
+applications to support easy error management and responses
 
+Each exception easily marshals to JSON based on the RFC9457 spec for use in api
```
Comment (JoseKilo) [inline README.md:12]: link to the spec ?

File: web_error/error.py [hunk 2]
```diff
@@ -1,7 +1,15 @@
...
+
+https://www.rfc-editor.org/rfc/rfc9457.html
+"""
+
 from __future__ import annotations
 
+import re
 import typing
 
+convert_re = re.compile(r"(?<!^)(?=[A-Z])")
```
Comment (JoseKilo) [inline web_error/error.py:11]: uppercase ? it's a constant

File: web_error/error.py [hunk 3]
```diff
@@ -12,51 +20,69 @@ class HttpException(Exception):  # noqa: N818
...
-            status=status,
-        )
-
 
 class HttpCodeException(HttpException):
-    status = None
     code = None
-    message = None
+    title = None
+    status = None
```
Comment (JoseKilo) [inline web_error/error.py:82]: `title` and `status` can't be `None` according to `HttpException`'s typing

**Review by JoseKilo — APPROVED** [review]
LGTM ! :guitar:

File: web_error/constant.py [hunk 4]
```diff
@@ -1,4 +1 @@
-BAD_REQUEST = 400
-NOT_FOUND = 404
-VALIDATION_ERROR = 422
-SERVER_ERROR = 500
+from http import HTTPStatus  # noqa: F401
```
[the remark this answers is not in the document]
Comment (EdgyEdgemond) [reply web_error/constant.py:1]: not really, just cos these constants were there and didn't give a full spec. Happy to drop as we are doing a minor (backwards breaking) release anyway

File: web_error/error.py [hunk 5]
```diff
@@ -12,51 +20,69 @@ class HttpException(Exception):  # noqa: N818
...
-        self.debug_message = debug_message
-
-    def marshal(self: typing.Self, *, strip_debug: bool = False) -> dict[str, str]:
+        self.extras = kwargs
+
+    @property
+    def type(self: typing.Self) -> str:
+        type_ = self.__class__.__name__.replace("Error", "")
+        type_ = convert_re.sub("-", type_).lower()
+        return self._code if self._code else type_
```
[the remark this answers is not in the document]
Comment (EdgyEdgemond) [reply web_error/error.py:40]: i'd say no, too much danger of that being non unique. Spec says type MUST be unique

File: web_error/error.py (hunk 3 above)
Comment (EdgyEdgemond) [reply web_error/error.py:82]: yeah, up to the base class to inject them.

Comment (EdgyEdgemond) [reply web_error/error.py:82]: No harm having a default for the subclass to replace though

**Pull request merged**